### PR TITLE
Support Mice in iPadOS 13.4+

### PR DESCRIPTION
**Pull request by gw280**
## Description

Adds support for iOS 13.4 mouse support.

Redo of https://github.com/flutter/engine/pull/23362

## Related Issues

https://github.com/flutter/flutter/issues/54663
https://github.com/flutter/flutter/issues/55809

## Tests

I added the following tests:

testPanGestureRecognizer
testMouseSupport

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [contributor guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [C++, Objective-C, Java style guides] for the engine.
- [x] I read the [tree hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation.
- [x] All existing and new tests are passing.
- [x] I am willing to follow-up on review comments in a timely manner.

## Reviewer Checklist

- [ ] I have submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

## Breaking Change

Did any tests fail when you ran them? Please read [handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[contributor guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[tree hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[engine presubmit flakes form]: https://forms.gle/Wc1VyFRYJjQTH6w5A

### CODE REVIEW

**Comment by gw280** [comment]
This adds the header guard in FlutterViewController.h

File: shell/platform/darwin/ios/framework/Headers/FlutterViewController.h [hunk 1]
```diff
@@ -47,8 +47,13 @@ extern NSNotificationName const FlutterSemanticsUpdateNotification;
  * Dart-related state and asynchronous tasks when navigating back and forth between a
  * FlutterViewController and other `UIViewController`s.
  */
+#ifdef __IPHONE_13_4
```
Comment (gaaclarke) [inline shell/platform/darwin/ios/framework/Headers/FlutterViewController.h:50]: This is conditionally removing the FLUTTER_EXPORT.  If we want to go this route you should make the scope the the conditional compilation around the UIPointerInteractionDelegate.

**Review by gaaclarke — APPROVED** [review]
LGTM assuming you've tested it manually with the old SDK and you change the scope of the conditional compilation.

You might also get compilation errors about unrecognized selectors for the other mouse stuff too.  That's why it would be nice to test this locally.

File: shell/platform/darwin/ios/framework/Headers/FlutterViewController.h (hunk 1 above)
Comment (gw280) [reply shell/platform/darwin/ios/framework/Headers/FlutterViewController.h:50]: good catch, I'll fix that. We should be FLUTTER_EXPORTING in every case. My bad.

**Comment by gw280** [comment]
Quick update: @jmagman tested this locally with Xcode 11 and it worked for her, so I think we can go ahead and re-land this. I would have liked to have done a full CI dry-run on the Flutter side but apparently that's not doable yet - I've filed an issue to support that at https://github.com/flutter/flutter/issues/73893.

**Pull request merged**